File: doc-examples/example-java/src/main/java/example/azure/BlobServiceClientBuilderCustomizer.java
package example.azure;

import com.azure.core.http.policy.HttpPipelinePolicy;
import com.azure.storage.blob.BlobServiceClientBuilder;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import io.micronaut.core.annotation.NonNull;
import jakarta.inject.Singleton;

//tag::class[]
@Singleton
public class BlobServiceClientBuilderCustomizer implements BeanCreatedEventListener<BlobServiceClientBuilder> {

    @Override
    public BlobServiceClientBuilder onCreated(@NonNull BeanCreatedEvent<BlobServiceClientBuilder> event) {
        HttpPipelinePolicy noOp = (context, next) -> next.process();
        return event.getBean().addPolicy(noOp);
    }
}
//end::class[]

File: doc-examples/example-python/src/main/python/example/azure/BlobServiceClientBuilderCustomizer.py
from com.azure.core.http.policy import HttpPipelinePolicy
from com.azure.storage.blob import BlobServiceClientBuilder
from jakarta.inject import Singleton
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener


# tag::class[]
@Singleton
class BlobServiceClientBuilderCustomizer(BeanCreatedEventListener[BlobServiceClientBuilder]):

    def onCreated(self, event: BeanCreatedEvent[BlobServiceClientBuilder]) -> BlobServiceClientBuilder:
        no_op: HttpPipelinePolicy = lambda context, next: next.process()
        return event.getBean().addPolicy(no_op)
# end::class[]
